src/cloudai/models/workload.py

```
@@ -191,3 +207,44 @@ def validate_agent_config(self) -> Self:
+        if unknown:
+            raise ValueError(f"env_params keys {unknown} are not cmd_args fields on {type(self.cmd_args).__name__}")
+
+        for name, spec in self.env_params.items():
+            if spec.weights is None:
+                continue
+            value = getattr(self.cmd_args, name, None)
+            if not isinstance(value, list) or len(value) < 2:
+                raise ValueError(
+                    f"env_params['{name}'] declares weights but cmd_args.{name} is not a candidate list "
+                    "(need a list of >= 2 values)"
+                )
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)